daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala

```diff
@@ -613,4 +604,176 @@ object Transaction {
+                  stakeholders1 == stakeholders2 &&
+                  key1 === key2 =>
+              loop(rest1, rest2, stack)
+            case (
+                Node.NodeFetch(
+                  coid1,
+                  templateId1,
+                  optLocation1 @ _,
+                  actingParties1,
+                  signatories1,
+                  stakeholders1,
+                  key1,
+                  byKey1 @ _,
+                ),
+                Node.NodeFetch(
+                  coid2,
+                  templateId2,
+                  optLocation2 @ _,
+                  actingParties2,
+                  signatories2,
+                  stakeholders2,
+                  key2,
+                  byKey2 @ _,
+                ))
+                if coid1 === coid2 &&
+                  templateId1 == templateId2 &&
+                  (actingParties1.isEmpty || actingParties1 == actingParties2) &&
+                  signatories1 == signatories2 &&
+                  stakeholders1 == stakeholders2 &&
+                  (key1.isEmpty || key1 === key2) =>
+              loop(rest1, rest2, stack)
+            case (
+                exe1 @ Node.NodeExercises(
+                  targetCoid1,
+                  templateId1,
+                  choiceId1,
+                  optLocation1 @ _,
+                  consuming1,
+                  actingParties1,
+                  chosenValue1,
+                  stakeholders1,
+                  signatories1,
+                  choiceObservers1,
+                  controllersDifferFromActors1,
+                  children1 @ _,
+                  exerciseResult1 @ _,
+                  key1,
+                  byKey1 @ _,
+                ),
+                exe2 @ Node.NodeExercises(
+                  targetCoid2,
+                  templateId2,
+                  choiceId2,
+                  optLocation2 @ _,
+                  consuming2,
+                  actingParties2,
+                  chosenValue2,
+                  stakeholders2,
+                  signatories2,
+                  choiceObservers2,
+                  controllersDifferFromActors2,
+                  children2 @ _,
+                  exerciseResult2 @ _,
+                  key2,
+                  byKey2 @ _,
+                ))
+                // results are checked after the children
+                if targetCoid1 === targetCoid2 &&
+                  templateId1 == templateId2 &&
+                  choiceId1 == choiceId2 &&
+                  consuming1 == consuming2 &&
+                  actingParties1 == actingParties2 &&
+                  chosenValue1 === chosenValue2 &&
+                  stakeholders1 == stakeholders2 &&
+                  signatories1 == signatories2 &&
+                  choiceObservers1 == choiceObservers2 &&
+                  controllersDifferFromActors1 == controllersDifferFromActors2 &&
+                  (key1.isEmpty || key1 === key2) =>
```

Comment:
This is again asymmetric.

Reply:
same thing than above, it is for backward compatibility.